### ISSUE TITLE
Forbid negated conditions

### ISSUE THREAD

**Issue by kxepal** [wemake-services/wemake-python-styleguide#438]
# Rule request

This rule should forbid conditions where double negations presents. 

## Thesis

Bad example
```
no_odds = all(i for i in range(10) if i % 2 == 0)
if not no_odds:
  do_a()
else:
  do_b()
```
`not no_odd` looks and reads ugly. Instead, we better write:
```
has_odds = any(i for i in range(10) if i % 2 == 1)
if has_odds:
  do_b()
else:
  do_a()
```
This inverts the if branches by making condition more readable.

Ok, that's a synthetic example I just imagined. Let's take few real ones:
https://github.com/prompt-toolkit/python-prompt-toolkit/blob/master/prompt_toolkit/key_binding/bindings/vi.py#L195
https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby/groupby.py#L886
https://github.com/cython/cython/blob/master/Cython/Compiler/Optimize.py#L4616
and so on. 

## Reasoning

Such expressions are hard to read and tends to have a better naming. Better naming simplifies readability and understanding things that happens. As long as understanding things is simple it's hard to make a mistake or easy to find the one.

Suggestions here could be to choose a better name for a variable _and_, probably, some branch reordering. This rule is couldn't be satisfied by find&replace strategy.

**Comment by sobolevn** [comment]
@kxepal you are ahead of my stuff. This is actually my "internal" backlog. Now that's official.

**Comment by kxepal** [comment]
Heh, sorry (: I recall this case after `not not` thing.

**Comment by sobolevn** [comment]
Reference: https://github.com/PyCQA/pylint/blob/master/pylint/checkers/refactoring.py#L1199

**Comment by sobolevn** [comment]
I rephrase a rule a little bit: do not start conditions with `not` if you have `else` clause.
Also, do not use `!=` for the same reason. 

So, instead of writing:

```python
if not some:
     ...
else: ...
```

**Comment by mikhail-akimov** [comment]
I`ll solve this issue!

**Comment by mikhail-akimov** [comment]
How do I get started more correctly? To use my local copy from preveous issue as a base (to update it somehow?) or to fork a new one?

**Comment by sobolevn** [comment]
Create a new one. It would be way easier.

To start you should consider several things:
1. Understand what a negated condition is, and what is not
2. Create a new violation in best practices
3. Create a new visitor called `NegatedConditionVisitor`, place it inside https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/visitors/ast/conditions.py
4. Write the logic 
5. Test it!
6. Submit for review

Feel free to ask any questions. 👍

**Comment by mikhail-akimov** [comment]
Thank you!

**Comment by sobolevn** [comment]
Closed via https://github.com/wemake-services/wemake-python-styleguide/pull/505
Refs #506
